### DISCUSSION TITLE
How to plot GND as a function of the grain size?

### DISCUSSION THREAD

**Discussion by jucego333** [mtex-toolbox/mtex#775] [Ask Anything]
Dear colleagues and Professors.

I am pretty new to MTEX.  I would like to know how to plot the GND (y-axis) as a function of the grain size distribution of my EBSD data? (x-axis). This figure will look like a scatter plot. 
![GND_example](https://user-images.githubusercontent.com/78469467/107925144-fd530080-6fae-11eb-8608-74cc0eae076a.png)

**Comment by kilir** [comment]
Hi,
you need to average/sum (or whatever you think is appropriate for each grain) your GND since it is computed for each pixel. Search the forum, this has been asked a couple of times, see `grainMean()`. Then you can simply plot the to lists of data using e.g. `scatter()`.
Cheers,
Rüdiger

**Reply by jucego333** [reply]
Thank you for your soon reply. I was reading the documentation about 'grainMean()' and checking the forum (I found only one case similar). However, I am still confused.  According to the documentation https://mtex-toolbox.github.io/GND.html, GND in simple terms, can be treated as column vector  gnd_tot = factor*sum(abs(rho .* dSRot.u),2). After that we need to approximate the GND at each grain, then we can use the function meanProp = grainMean(ebsd, prop, grains, method), that in our case should be something like GND_average=grainMean(ebsd, gnd_tot, ???, ???).  My question is : what is the variable "grains" in this case?  I will deeply appreciate any guidelines and comments.

**Reply by kilir** [reply]
Hi,
ebsd and grains are linked via the grainId. If you want to ensure that the list of the averaged property has the same order as the grains (or is only specified for a subset of grains) you are averaging over, use the `grains` argument.
Hope that helps.
Cheers,
Rüdiger

**Reply by jucego333** [reply]
Hi, Kilir

Is right the following lines of code to get the plot stated above?

gnd_tot = factor*sum(abs(rho .* dSRot.u),2);
meanProp = grainMean(ebsd,gnd_tot,grains);
scatter (grains.area, meanProp)

Best regards

**Comment by lfgmorales** [comment]
in addition to what rüdiger just said, the variable 'grains' is calculated from your EBSD map, like here


https://mtex-toolbox.github.io/GrainReconstruction.html



cheers



Luiz


Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________
